cellxgene_schema_cli/cellxgene_schema/utils.py

```diff
@@ -146,3 +148,16 @@ def _enforce_canonical_format(df):
     if adata.raw:
         logger.info("enforce canonical format in raw.X")
         _enforce_canonical_format(adata.raw)
+
+
+def get_hash_digest_column(dataframe):
+    """
+    Get column with hash digest for each row in dataframe.
+    """
+
+    return (
```

Comment:
this code - as currently used - assumes the index is unique (which is not something AnnData requires).  For sanity's sake, you might want to assert `datafame.index.is_unique` here.

Reply:
we validate [uniqueness](https://github.com/chanzuckerberg/single-cell-curation/blob/main/cellxgene_schema_cli/cellxgene_schema/validate.py#L356) of [indices](https://github.com/chanzuckerberg/single-cell-curation/blob/main/cellxgene_schema_cli/cellxgene_schema/validate.py#L628) as part of the validate step, which runs before the add-labels function where this is called! 

Though, probably would still be useful to make it explicit here since it's tucked in a util function rather than in AnnDataLabelAppender

Reply:
Yep, I saw that, but there is nothing that prevents a misconfiguration for `obs.index`, which would be unfortunate and lead to non-unique IDs.

Not sure how you want to handle this, but given that AnnData allows duplicate indices, and we do _not_ allow them, a bit more belt-and-suspenders might be safe...

Reply:
I like Bruce's suggestion of including an `assert`

Reply:
I agree, add the `assert`